storage/client/client.go

```diff
@@ -491,6 +491,59 @@ func (c *StorageClient) Events(ctx context.Context, p apiTypes.GetConsensusEvent
 	return &es, nil
 }
 
+func (c *StorageClient) RoothashMessages(ctx context.Context, p apiTypes.GetConsensusRoothashMessagesParams) (*apiTypes.RoothashMessageList, error) {
+	if p.Round != nil && p.Runtime == nil {
+		return nil, fmt.Errorf("must specify runtime with round")
+	}
+	res, err := c.withTotalCount(
+		ctx,
+		queries.RoothashMessages,
+		p.Runtime,
+		p.Round,
+		p.Type,
+		p.Rel,
+		p.Limit,
+		p.Offset,
+	)
+	if err != nil {
+		return nil, wrapError(err)
+	}
+	defer res.rows.Close()
+
+	ms := RoothashMessageList{
+		RoothashMessages:    []RoothashMessage{},
+		TotalCount:          res.totalCount,
+		IsTotalCountClipped: res.isTotalCountClipped,
+	}
+
+	for res.rows.Next() {
+		var m RoothashMessage
+		var resultCBOR *[]byte
+		if err := res.rows.Scan(
+			&m.Runtime,
+			&m.Round,
+			&m.Index,
+			&m.Type,
+			&m.Body,
+			&m.ErrorModule,
+			&m.ErrorCode,
+			&resultCBOR,
```

Comment:
would it be better to store the result as a json object in the db during analysis time?

Reply:
we need the message type to do the conversion, and we get the type and result at different times during analysis, potentially even in reverse order with fast sync. it would be nice, but it's kinda weird to implement.